### PR TITLE
Add endpoints

### PR DESCRIPTION
**Pull request by allthatilk**
Adding new datastore API endpoints to the Query Builder

### CODE REVIEW

**Comment by allthatilk** [comment]
Ugh, this code is so WET but no time for a full overhaul right now.

File: templates/index.html [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M" crossorigin="anonymous">
     <link href="static/css/custom.css" type="text/css" rel="stylesheet">
     <link href="static/css/chosen.css" type="text/css" rel="stylesheet">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.4.1/css/bootstrap-datepicker3.css"/>
```
Comment (andylolz) [inline templates/index.html]: You shouldn’t close a link tag, because it’s a void element:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link

File: index.php [hunk 2]
```diff
@@ -63,19 +87,37 @@
   if (isset($_POST["entry_1922375458"])) { //organisations
     $requested_orgs = filter_var_array($_POST["entry_1922375458"], FILTER_SANITIZE_STRING);
     foreach ($requested_orgs as $requested_org) {
-      if (in_array($requested_org, $allowed_orgs) && !empty($requested_org) ) { //!!!!FIX ME!!!!
+      if (in_array($requested_org, $allowed_orgs) && !empty($requested_org) ) { //!!!!FIX ME!!!!I would but I don't know how you're broken good sir...
```
Comment (andylolz) [inline index.php]: A bunch of these FIX MEs were introduced in the very first commit:
https://github.com/IATI/IATI-Query-Builder/commit/530c4c51#diff-828e0013b8f3bc1bb22b4f57172b019dR35

In all cases, the `$allowed` arrays (the whitelists) were empty, and the `in_array` conditions were flipped. So I guess the FIX MEs referred to that.

It looks like this has since been fixed (e.g. `$allowed_orgs` is now populated, and is actually used as a whitelist), and while all the other FIX MEs were removed, this one wasn’t.

In short: I think it’s safe to remove.

File: index.php [hunk 3]
```diff
@@ -63,19 +87,37 @@
...
         $orgs[] = $requested_org;
       }
     }
   }
-
+  //print_r($allowed_orgs);
+  if (isset($_POST["transaction_provider_org"])) { //organisations
+    $requested_orgs = filter_var_array($_POST["transaction_provider_org"], FILTER_SANITIZE_STRING);
+    foreach ($requested_orgs as $requested_org) {
+      if (in_array($requested_org, $allowed_orgs) && !empty($requested_org) ) { //!!!!FIX ME!!!!I would but I don't know how you're broken good sir...
```
Comment (andylolz) [inline index.php]: See [this comment](https://github.com/IATI/IATI-Query-Builder/pull/71/files#r140635099). I think it’s safe to remove this FIX ME.

File: templates/index.html [hunk 4]
```diff
@@ -215,6 +216,71 @@ <h1 dir="ltr">IATI Data Store CSV Query Builder (Alpha Version)</h1>
...
+                <input type="radio" name="before_end" id="before_end" value="on" checked> before
+              </label>
+              <label class="btn btn-primary">
+                <input type="radio" name="before_end" id="before_end" value=""> after
+              </label>
+            </div>
+            <div class="input-group date" data-provide="datepicker" data-date-format="yyyy-mm-dd">
+              <input placeholder="yyyy-mm-dd" type="date" id="end_date" name="end_date">
+                <div class="input-group-addon"/></div>
+              </input>
```
Comment (andylolz) [inline templates/index.html]: Neither datepicker is repopulated by PHP, like the other fields.

File: templates/index.html (hunk 1 above)
Comment (andylolz) [inline templates/index.html]: This date picker [doesn’t support bootstrap4 yet](https://github.com/uxsolutions/bootstrap-datepicker/issues/1670#issuecomment-329536391). I wonder if there’s one that does?

**Comment by andylolz** [comment]
I’ve just realised this has an `incomplete` label, so I’m premature with comments! But it’s possible they are useful anyway, so I’ll leave them.

**Comment by allthatilk** [comment]
Thanks @andylolz for spotting these =)

File: templates/index.html (hunk 1 above)
Comment (andylolz) [reply templates/index.html]: If not… I wonder if there’s a case for using bootstrap3 instead? v4 is very newly out of alpha, and I am not sure it is terribly well-supported by these sorts of extensions yet.

File: templates/index.html [hunk 5]
```diff
@@ -157,7 +158,7 @@ <h1 dir="ltr">IATI Data Store CSV Query Builder (Alpha Version)</h1>
             </label>
             <div class="clearfix"></div>
             <select data-placeholder="Choose a reporting organisation..." multiple class="chosen-select form-control" id="entry_1922375458[]" name="entry.1922375458[]">
-              {{ include('include/reporting_org.html') }}
+              {{ include ('include/reporting_org.html') }}
```
[the remark this answers is not in the document]
Comment (allthatilk) [reply templates/index.html]: done

File: templates/index.html (hunk 1 above)
Comment (allthatilk) [reply templates/index.html]: done

File: index.php (hunk 2 above)
Comment (allthatilk) [reply index.php]: Wizard

**Comment by allthatilk** [comment]
This work is being taken forward by #78
